htdocs/product/card.php

```diff
@@ -1944,7 +1944,45 @@
 				if ($object->isProduct() || !empty($conf->global->STOCK_SUPPORTS_SERVICES)) {
 					print '<tr><td>'.$langs->trans("ManageLotSerial").'</td><td>';
 					$statutarray = array('0' => $langs->trans("ProductStatusNotOnBatch"), '1' => $langs->trans("ProductStatusOnBatch"), '2' => $langs->trans("ProductStatusOnSerial"));
-					print $form->selectarray('status_batch', $statutarray, (GETPOSTISSET('status_batch') ? GETPOST('status_batch') : $object->status_batch));
+
+					print $form->selectarray('status_batch', $statutarray, $object->status_batch);
```

Comment:
On error, we should not lose the user initial delection, so can you restore the test on GETPOSTISSET in current line of code ?